ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/RequestDataStep.java

```diff
@@ -131,15 +134,36 @@ public CompletableFuture<List<Task<SnapDataRequest>>> requestStorage(
             (response, error) -> {
               if (response != null) {
                 downloadState.removeOutstandingTask(getStorageRangeTask);
-                for (int i = 0; i < response.slots().size(); i++) {
-                  final StorageRangeDataRequest request =
-                      (StorageRangeDataRequest) requestTasks.get(i).getData();
-                  request.setRootHash(blockHeader.getStateRoot());
-                  request.addResponse(
-                      downloadState,
-                      worldStateProofProvider,
-                      response.slots().get(i),
-                      i < response.slots().size() - 1 ? new ArrayDeque<>() : response.proofs());
+                final ArrayDeque<NavigableMap<Bytes32, Bytes>> slots = new ArrayDeque<>();
+                // Check if we have an empty range
+
+                /*
+                 * Checks if the response represents an "empty range".
+                 *
+                 * An "empty range" is defined as a response where either no slots are present,
+                 * or the first slot is empty, but at least one proof exists
+                 */
+                try {
+                  final boolean isEmptyRange =
+                      (response.slots().isEmpty() || response.slots().get(0).isEmpty())
+                          && !response.proofs().isEmpty();
```

Comment:
should this be ((A || B) && !C)
or
(A || (B & !C))

does it make sense for the response.slots to be empty if response.proofs is not empty

Reply:
yes, in that case it would be a proof of exclusion 👍

Reply:
|| takes precedent over &&, so this should be ((A ||B) && !C).  🙏

Reply:
added a suggested rewrite for the comment